xs/src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -54,7 +71,7 @@ PerimeterGenerator::process()
     for (Surfaces::const_iterator surface = this->slices->surfaces.begin();
         surface != this->slices->surfaces.end(); ++surface) {
         // detect how many perimeters must be generated for this island
-        const int loop_number = this->config->perimeters + surface->extra_perimeters -1;  // 0-indexed loops
+        const int loop_number = num_loops(this->config->perimeters, surface->extra_perimeters);//this->config->perimeters + surface->extra_perimeters-1;  // 0-indexed loops
```

Comment:
Just a note about semantics: `loop_number` holds the number of loops diminuished by 1. This is related to the implementation of the subsequent code, and I admit the variable could be named in a better way. But when you expose a `num_loops()` method, it should output the *actual* number of loops. Then you can subtract 1 in the caller method if it's needed by its implementation.

By the way, I don't see the need for a separate method just for this calculation. I'd embed the logic in the main method.